client/asset/btc/btc.go

```diff
@@ -3190,6 +3190,10 @@ func preAccelerate(btc *baseWallet, swapCoins, accelerationCoins []dex.Bytes, ch
 // are NOT manually unlocked because they're auto-unlocked when the transaction
 // is broadcasted.
 func (btc *baseWallet) Swap(swaps *asset.Swaps) ([]asset.Receipt, asset.Coin, uint64, error) {
+	if swaps.FeeRate == 0 {
+		return nil, nil, 0, fmt.Errorf("cannot send swap with with zero fee rate")
+	}
```

Comment:
I wish I remembered better the details of #1587, but I think it was an issue on the server. In that case, it would be fine to check just once at https://github.com/decred/dcrdex/blob/69ace7bdafad0a8b70714be4e5b7153f4c83efed/client/core/trade.go#L2329-L2330

```Go
if highestFeeRate == 0 {
    return fmt.Errorf(...)
}
```

Reply:
Yes, we can do that and save a few LOC. I initially assumed the check would be better at the asset level.

Reply:
I don't have a problem with it in both places TBH, but core also shouldn't request impossible swaps.

Reply:
> I don't have a problem with it in both places TBH

I left the check at the asset level as an extra precaution, since it is very crucial to the swap process. Or do we really want to remove the check from the asset level? @chappjc